scala/src/main/org/apache/spark/streaming/api/csharp/CSharpDStream.scala

```diff
@@ -102,7 +102,7 @@ class CSharpDStream(
   override def slideDuration: Duration = parent.slideDuration
 
   override def compute(validTime: Time): Option[RDD[Array[Byte]]] = {
-    val rdd = parent.compute(validTime)
+    val rdd = parent.getOrCompute(validTime)
```

Comment:
it's a bug I think.
getOrCompute put generated RDDs in a HashMap for future reference including checkpoint.
double checked python did same thing.
not sure why it was compute.